src/ndata.c

```diff
@@ -118,28 +95,61 @@ int ndata_check( const char* path )
  *    @param path Path to set.
  *    @return 0 on success.
  */
-int ndata_setPath( const char* path )
+int ndata_setPath( const char *path )
 {
    int len;
+   char *buf;
 
-   free(ndata_filename);
-   free(ndata_dirname);
-   ndata_filename = NULL;
-   ndata_dirname  = NULL;
+   if ( ndata_filename != NULL ) {
+      free( ndata_filename );
+      ndata_filename = NULL;
+   }
 
-   if (path == NULL)
-      return 0;
-   else if (nfile_dirExists(path)) {
-      len = strlen(path);
-      ndata_dirname = strdup(path);
-      if (nfile_isSeparator(ndata_dirname[len - 1]))
-         ndata_dirname[len - 1] = '\0';
+   if ( path != NULL && ndata_isndata( path ) ) {
+      len            = strlen( path );
+      ndata_filename = strdup( path );
+      if ( nfile_isSeparator( ndata_filename[ len - 1 ] ) )
+         ndata_filename[ len - 1 ] = '\0';
+      ndata_source = NDATA_SRC_USER;
    }
-   else if (nfile_fileExists(path)) {
-      char *tmp = strdup(path);
-      ndata_filename = strdup(path);
-      ndata_dirname  = nfile_dirname(tmp);
+   else {
+      ndata_source = NDATA_SRC_SEARCH_START;
+
+      switch ( ndata_source ) {
+      case NDATA_SRC_CWD:
+         ndata_filename = ndata_findInDir( "." );
+         if ( ndata_filename != NULL ) {
+            ndata_source = NDATA_SRC_CWD;
+            break;
+         }
+         __attribute__( ( fallthrough ) );
```

Comment:
Apparently these fallthrough are breaking the mac os x compiles. Might be worth doing something like https://stackoverflow.com/questions/45349079/how-to-use-attribute-fallthrough-correctly-in-gcc instead. See also https://clang.llvm.org/docs/AttributeReference.html#fallthrough

Reply:
Looks like it doesn't like the fallthrough attribute on case with only a comment. The rest are just warns. I'll make sure all warns are cleaned up in the Meson builds